### DISCUSSION TITLE
Any thoughts on a future docusaurus-plugin-doxygen?

### DISCUSSION THREAD

**Discussion by ilg-ul** [doxygen/doxygen#11215] [Ideas]
I recently migrated one of my TypeScript projects to [Docusaurus](https://docusaurus.io) and used [docusaurus-plugin-typedoc](https://typedoc-plugin-markdown.org/plugins/docusaurus) for the TypeDoc integration (the result is at https://xpack.github.io/logger-ts/).

Next I want to migrate some C/C++ projects sites to Docusaurus.

I could not find any project aiming to integrate the Doxygen reference pages into Docusaurus, so I'm considering writing a `docusaurus-plugin-doxygen` to do this.

The plan is to run Doxygen in an external separate step, to output XML files that will remain persistent in the project, then have the Docusaurus plugin parse the XML files and generate the React components for the Docusaurus pages.

Any thoughts on such an approach? 

Anyone else interested in such a solution?

**Comment by paule32** [comment]
I can think over such of things to implement it at my current project: https://github.com/paule32/HelpNDocTools
The time will show it...

**Reply by ilg-ul** [reply]
From what I see now, your project is a Python application.

Docusaurus is a TypeScript/JavaScript application, and Docusaurus plugins should also be TypeScript/JavaScript Node.js modules. I see no relationship between the two projects.

**Reply by paule32** [reply]
I thinking over such things.
I never write, that I support it.
And yes, the project is a Python Application.

What I can imagine is, that the PlugIn can be inserted/merged into the Python Project Application.
I don't much over TypeScript.
But I thinking, that such things like JavaScript could be implemented in such a way, that the user is provided with a Interface.
And the leading Development Language on my side is Python.

Not more, not lesser...
